### PR TITLE
Turn tables dropped or excluded from the publication into normal tables

### PR DESCRIPTION
**Pull request by seut**
## Summary of the changes / Why this improves CrateDB

Tables dropped on the publisher or excluded from the publication must not be dropped on the subscriber but should turn into normal read-write tables.

## Checklist

 - [x] Added an entry in `CHANGES.txt` for user facing changes
 - [x] Updated documentation & `sql_features` table for user facing changes
 - [x] Touched code is covered by tests
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

File: server/src/test/java/io/crate/integrationtests/MetadataTrackerITest.java [hunk 1]
```diff
@@ -372,6 +310,41 @@ public void test_subscribing_to_the_own_tables_on_the_same_cluster() throws Exce
 
     }
 
+    @Test
+    public void test_alter_publication_drop_table_disables_table_replication() throws Exception {
+        executeOnPublisher("CREATE TABLE t1 (id INT) WITH(" + defaultTableSettings() + ")");
+        executeOnPublisher("INSERT INTO t1 (id) VALUES (1), (2)");
```
Comment (BaurzhanSakhariev) [inline server/src/test/java/io/crate/integrationtests/MetadataTrackerITest.java:316]: can we add third partitioned table and also exclude it - just to cover all cases?

File: server/src/main/java/io/crate/replication/logical/MetadataTracker.java [hunk 2]
```diff
@@ -460,23 +459,21 @@ static RestoreDiff getRestoreDiff(Subscription subscription,
...
         HashSet<RelationName> changedRelations = new HashSet<>();
-        HashSet<Index> indicesToRemove = new HashSet<>();
-        ArrayList<String> templatesToRemove = new ArrayList<>();
+        HashSet<Index> partitionsToRemove = new HashSet<>();
         Metadata subscriberMetadata = subscriberClusterState.metadata();
         for (var relationName : subscription.relations().keySet()) {
-            var publisherRelationMetadata = response.relationsInPublications().get(relationName);
-            var possibleTemplateName = PartitionName.templateName(relationName.schema(), relationName.name());
-            var isPartitioned = subscriberMetadata.templates().get(possibleTemplateName) != null;
-            boolean partitionDisappeared = isPartitioned && (publisherRelationMetadata == null || publisherRelationMetadata.template() == null);
```
Comment (BaurzhanSakhariev) [inline server/src/main/java/io/crate/replication/logical/MetadataTracker.java:475]: just to be sure - it looks like we don't use `publisherRelationMetadata.template()`anymore (and in general removed templates logic), so can drop it from `PublicationStateAction.Response`?

File: server/src/main/java/io/crate/replication/logical/MetadataTracker.java [hunk 3]
```diff
@@ -522,6 +508,12 @@ private ClusterState removeDroppedTablesOrPartitions(String subscriptionName,
                     relations.put(relationName, state);
                 }
             }
+            updatedClusterState = DropSubscriptionAction.removeSubscriptionSetting(
```
Comment (BaurzhanSakhariev) [inline server/src/main/java/io/crate/replication/logical/MetadataTracker.java:511]: I guess this is for stopping tracking of those relations? Can we add tracker not alive check in the end of the test?

Comment (seut) [reply server/src/main/java/io/crate/replication/logical/MetadataTracker.java:511]: No the tracking of the publication will continue.

**Review by BaurzhanSakhariev — APPROVED** [review]
👍

**Comment by seut** [comment]
retest this please

**Pull request merged**